File: app/src/main/java/utils/MediaPlayerManager.java
package utils;

import android.content.Context;
import android.media.MediaPlayer;

import com.adaca.memoria.R;

public class MediaPlayerManager {
    public final MediaPlayer hitFX, endFX, music;

    private float fxVolume, musicVolume;

    private static MediaPlayerManager instance = null;

    public MediaPlayerManager(Context context) {
        endFX = MediaPlayer.create(context, R.raw.fim);
        hitFX = MediaPlayer.create(context, R.raw.tumm_c5);
        music = MediaPlayer.create(context, R.raw.plancton);

        music.seekTo(0);
        music.setLooping(true);
    }

    public static MediaPlayerManager getInstance(Context context) {
        if (instance == null) {
            instance = new MediaPlayerManager(context);
        }

        return instance;
    }

    public static void resetInstance() {
        instance = null;
    }

    public void setMediaPlayersVolume(boolean toggleButtonIsChecked) {
        if (!toggleButtonIsChecked) {
            endFX.setVolume(fxVolume, fxVolume);
            music.setVolume(musicVolume, musicVolume);
            hitFX.setVolume(fxVolume, fxVolume);
        } else {
            endFX.setVolume(0, 0);
            music.setVolume(0, 0);
            hitFX.setVolume(0, 0);
        }
    }

    public float getFxVolume() {
        return this.fxVolume;
    }

    public void setFxVolume(float fxVolume) {
        this.fxVolume = fxVolume;
    }

    public float getMusicVolume() {
        return this.musicVolume;
    }

    public void setMusicVolume(float musicVolume) {
        this.musicVolume = musicVolume;
    }
}

File: app/src/main/kotlin/utils/MediaPlayerManager.kt
package utils

import android.content.Context
import android.media.MediaPlayer
import com.adaca.memoria.R

class MediaPlayerManager private constructor(context: Context) {
    val endFX: MediaPlayer = MediaPlayer.create(context, R.raw.fim)
    val hitFX: MediaPlayer = MediaPlayer.create(context, R.raw.tumm_c5)
    val music: MediaPlayer = MediaPlayer.create(context, R.raw.plancton).apply {
        seekTo(0)
        isLooping = true
    }
    var fxVolume = 0f
    var musicVolume = 0f

    fun setMediaPlayersVolume(toggleButtonIsChecked: Boolean) {
        if (!toggleButtonIsChecked) {
            endFX.setVolume(fxVolume, fxVolume)
            music.setVolume(musicVolume, musicVolume)
            hitFX.setVolume(fxVolume, fxVolume)
        } else {
            endFX.setVolume(0f, 0f)
            music.setVolume(0f, 0f)
            hitFX.setVolume(0f, 0f)
        }
    }

    companion object {
        @Volatile
        private var instance: MediaPlayerManager? = null

        fun getInstance(context: Context): MediaPlayerManager {
            return instance ?: synchronized(this) {
                instance ?: MediaPlayerManager(context).also { instance = it }
            }
        }

        fun resetInstance() {
            instance = null
        }
    }
}